sql/sqlliveness/slstorage: TestStorage failed

[(sql/sqlliveness/slstorage).TestStorage failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2394145&tab=buildLog) on [master@6184870a438ae34afbcf29dda5452345dc7587d3](https://github.com/cockroachdb/cockroach/commits/6184870a438ae34afbcf29dda5452345dc7587d3):

```
=== RUN   TestStorage
--- FAIL: TestStorage (0.43s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStorage365164136
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestStorage/delete-update
    --- FAIL: TestStorage/delete-update (0.02s)
        slstorage_test.go:117: 
            	Error Trace:	slstorage_test.go:117
            	            				soon.go:47
            	            				retry.go:197
            	            				soon.go:53
            	            				soon.go:34
            	            				slstorage_test.go:112
            	Error:      	Not equal: 
            	            	expected: []time.Time{time.Time{wall:0x0, ext:63082281620, loc:(*time.Location)(nil)}}
            	            	actual  : []time.Time{time.Time{wall:0x5492df7, ext:63082281617, loc:(*time.Location)(nil)}}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -2,4 +2,4 @@
            	            	  (time.Time) {
            	            	-  wall: (uint64) 0,
            	            	-  ext: (int64) 63082281620,
            	            	+  wall: (uint64) 88681975,
            	            	+  ext: (int64) 63082281617,
            	            	   loc: (*time.Location)(<nil>)
            	Test:       	TestStorage/delete-update

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestStorage PKG=./pkg/sql/sqlliveness/slstorage TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStorage.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/sqlliveness/slstorage/slstorage_test.go
+++ b/pkg/sql/sqlliveness/slstorage/slstorage_test.go
@@ -68,12 +68,12 @@ func TestStorage(t *testing.T) {
 		stopper := stop.NewStopper()
 		storage := slstorage.NewTestingStorage(stopper, clock, kvDB, ie, settings,
 			dbName, timeSource.NewTimer)
-		storage.Start(ctx)
 		return clock, timeSource, settings, stopper, storage
 	}
 
 	t.Run("basic-insert-is-alive", func(t *testing.T) {
 		clock, _, _, stopper, storage := setup(t)
+		storage.Start(ctx)
 		defer stopper.Stop(ctx)
 
 		exp := clock.Now().Add(time.Second.Nanoseconds(), 0)
@@ -219,6 +219,7 @@ func TestStorage(t *testing.T) {
 	t.Run("delete-expired-on-is-alive", func(t *testing.T) {
 		clock, timeSource, _, stopper, storage := setup(t)
 		defer stopper.Stop(ctx)
+		storage.Start(ctx)
 
 		exp := clock.Now().Add(time.Second.Nanoseconds(), 0)
 		const id = "asdf"
@@ -266,7 +267,6 @@ func TestStorage(t *testing.T) {
 		_, timeSource, settings, stopper, storage := setup(t)
 		defer stopper.Stop(ctx)
 		storage.Start(ctx)
-
 		waitForGCTimer := func() (timer time.Time) {
 			testutils.SucceedsSoon(t, func() error {
 				timers := timeSource.Timers()
@@ -285,6 +285,7 @@ func TestStorage(t *testing.T) {
 		}
 		jitter := slstorage.GCJitter.Get(&settings.SV)
 		interval := slstorage.GCInterval.Get(&settings.SV)
+		storage.Start(ctx)
 		minTime := t0.Add(time.Duration((1 - jitter) * float64(interval.Nanoseconds()) * N))
 		maxTime := t0.Add(time.Duration((1 + jitter) * float64(interval.Nanoseconds()) * N))
 		noJitterTime := t0.Add(interval * N)

